tests/components/kodi/test_binary_sensor.py

```diff
@@ -0,0 +1,78 @@
+"""The tests for Kodi binary sensor platform."""
+import logging
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_component import async_update_entity
+
+from . import init_integration
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "element, name",
+    [("screensaver", "Screensaver"), ("energy_saving", "Energy saving")],
+)
+async def test_binary_sensor_initial_states(hass: HomeAssistant, element, name):
+    """Test for binary sensor values."""
+    entry = await init_integration(hass)
+
+    state = hass.states.get(f"binary_sensor.{entry.data['name']}_{element}")
+    assert state.state == STATE_OFF
+    assert state.attributes["friendly_name"] == f"{entry.data['name']} {name}"
+
+
+@pytest.mark.parametrize("element", ["screensaver", "energy_saving"])
+async def test_binary_sensor_statechange(hass: HomeAssistant, element):
+    """Test the event."""
+    entry = await init_integration(hass)
+
+    state = hass.states.get(f"binary_sensor.{entry.data['name']}_{element}")
+    assert state.state == STATE_OFF
+
+    hass.states.async_set(f"binary_sensor.{entry.data['name']}_{element}", STATE_ON)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(f"binary_sensor.{entry.data['name']}_{element}")
+    assert state.state == STATE_ON
+
+
+@pytest.mark.parametrize(
+    "element, boolean",
+    [
+        ("screensaver", "System.ScreenSaverActive"),
+        ("energy_saving", "System.DPMSActive"),
+    ],
+)
+async def test_binary_sensor_update(hass: HomeAssistant, element, boolean):
```

Comment:
Maybe also test updates through a callback?

Reply:
😇 well I wanted and read and tried some things but I couldn't figure out how to get to them. The call itself originates in the jsonrpc module. I tried to figure out how to get the KodiBinaryEntity instances to call them directly. Can you give me a idea?

Reply:
you can patch the various registration properties, capture the callbacks that are being passed, and then call them

Reply:
Still, after your keywords, searching and reading, I don't see light at the end of tunnel. I know with patch you can guide the test subject to a certain state. In this case I need to make `init_integration` to actually use a websocket connection, and connect it so that the binary sensors register their callbacks. This registration I need to capture with something like [this (call_args)](https://docs.python.org/3/library/unittest.mock-examples.html#mocking-unbound-methods). Am I anywhere close to what you have in mind?

Reply:
Here's an example of how I patch a registration function:
https://github.com/home-assistant/core/blob/fc472eb040892ac9ab13a9a7ee6d2ae5bfecce66/tests/components/risco/test_binary_sensor.py#L130

(in Kodi's case it's a property, so you might have to use `PropertyMock`).
This patch is passed as a fixture to tests, and here's how I obtain the callback:
https://github.com/home-assistant/core/blob/fc472eb040892ac9ab13a9a7ee6d2ae5bfecce66/tests/components/risco/test_binary_sensor.py#L138

Reply:
Your hint really helped a lot, as usual. Thanks.
I'm setting up the platform and patch the registration function for callbacks in the latest commit.